node/relay.go

```diff
@@ -0,0 +1,88 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"go.uber.org/zap"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/config"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/metrics"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+	"github.com/spacemeshos/go-spacemesh/p2p/handshake"
+)
+
+var relayAddrInfoCh chan peer.AddrInfo // used for testing
+
+func runRelay(ctx context.Context, cfg *config.Config) error {
+	ctx, cancel := signal.NotifyContext(ctx, os.Interrupt, syscall.SIGTERM)
+	defer cancel()
+	p2pCfg := cfg.P2P
+	p2pCfg.DataDir = filepath.Join(cfg.DataDir(), "p2p")
+	p2pCfg.DisablePubSub = true
+	p2pCfg.Relay = true
+	p2pCfg.RelayServer.Enable = true
+
+	lvl := zap.NewAtomicLevel()
+	loggers, err := decodeLoggers(cfg.LOGGING)
+	if err != nil {
+		return fmt.Errorf("failed to decode loggers: %w", err)
+	}
+	level, ok := loggers[P2PLogger]
+	if ok {
+		if err := lvl.UnmarshalText([]byte(level)); err != nil {
+			return fmt.Errorf("cannot parse logging for %v: %w", P2PLogger, err)
+		}
+	} else {
+		lvl.SetLevel(log.DefaultLevel())
+	}
+
+	p2pCfg.LogLevel = lvl.Level()
+	p2pLog := log.NewWithLevel("node", zap.NewAtomicLevelAt(zap.DebugLevel)).WithName(P2PLogger)
```

Comment:
I think this is also repeated code from the `node` command. Maybe we can make the methods that initialize the logger for the node on the `App` type functions that can be called in both commands?

Reply:
I'd prefer not to use `App` for relay-only mode b/c it carries with itself a lot of functionality which is not used in this case, and I hope this relay-only mode will be a temporary feature to be removed later, when we have enough normal nodes acting as relays. But I refactored it a bit to reduce code duplication